batch_daily/workflows.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+from datetime import datetime, timedelta
+
+from dataclasses import dataclass
+import time
+from typing import Any, Dict, Optional
+
+from temporalio import workflow
+from temporalio.common import RetryPolicy
+from temporalio.exceptions import ActivityError
+from temporalio.common import SearchAttributeKey
+
+with workflow.unsafe.imports_passed_through():
+    from batch_daily.activities import (
+        ListRecordActivityInput,
+        list_records,
+        ProcessRecordActivityInput,
+        process_record,
+    )
+
+TASK_QUEUE_NAME = "MY_TASK_QUEUE"
+
+
+@dataclass
+class RecordBatchProcessorWorkflowInput:
+    record_filter: str
+    day: Optional[str] = None
+
+
+@workflow.defn
+class RecordBatchProcessor:
+    @workflow.run
+    async def run(
+        self, workflow_input: RecordBatchProcessorWorkflowInput
+    ) -> Dict[str, Any]:
+        if workflow_input.day is None:
+            schedule_time = workflow.info().typed_search_attributes.get(
+                SearchAttributeKey.for_datetime("TemporalScheduledStartTime")
+            )
+            assert schedule_time is not None, "when not scheduled, day must be provided"
+            workflow_input.day = schedule_time.strftime("%Y-%m-%d")
+
+        print(f"starting RecordProcessor with {workflow_input}")
+
+        list_records_input = ListRecordActivityInput(
+            record_filter=workflow_input.record_filter, day=workflow_input.day
+        )
+
+        record_uri_list = await workflow.execute_activity(
+            list_records,
+            list_records_input,
+            start_to_close_timeout=timedelta(minutes=5),
+        )
+
+        task_list = []
+        async with asyncio.TaskGroup() as tg:
+            for key in record_uri_list:
```

Comment:
This is no good as it runs the risk of spamming the event history log. It should probably work similarly to the java example where it uses continue_as_new to reset the event history.